### PR TITLE
[AGW HA][proto] Define HAd protobuf for LTE AGW HA

### PR DESCRIPTION
**Pull request by mpgermano**
Signed-off-by: Michael Germano <mgermano@fb.com>

## Summary

This PR defines the protobuf to be used for the new lte HAd service. 

The idea with this proto is that backup instances in an AGW pool (i.e. cloud based AGWs) will call the 
defined RPC endpoint to fetch all relevant ENB states in order to determine which ENBs to offload back 
to the primary.

The logic for this RPC method should work something like this (open for discussion):

1. AWS backup AGW calls `GetEnodebOffloadState`
2. Orc8r fetches gatewayID of caller
3. Using gatewayID, fetch all gateway pool IDs that this gateway belongs to
4. For each pool, fetch all gateways in the pool:
   a. Skip any gateways whose mme relative capacity is lower than the calling gateway
   b. For each ENB connected to a gateway in the pool, query how many users are connected to the ENB
            i. If connected users > 0, set ENB state to PRIMARY_CONNECTED_AND_SERVING_UES
            ii. Otherwise, connected users == 0, set ENB state to PRIMARY_CONNECTED.
5. Return map of all ENB states. If an ENB is missing, it is considered to be a no-op. This will be the case for ENBs connected to the secondary whose primary is still offline.

### CODE REVIEW

**Comment by ulaskozat** [comment]
This summary looks good to me. So the implicit assumption is that an  AGW X is a backup to another AGW Y if and only if:
1. X is included in the pool that Y belongs to.
2. X has a relative MME capacity that is strictly less than that of Y.

File: lte/protos/had_orc8r.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+message GetEnodebOffloadStateResponse {
+    enum EnodebOffloadState {
+      NO_OP = 0;
+      PRIMARY_CONNECTED = 1;
+      PRIMARY_CONNECTED_AND_SERVING_UES = 2;
+    }
+    // Map from ENB SN to offload state
+    map<string, EnodebOffloadState> enodeb_offload_states = 1;
```
Comment (ulaskozat) [inline lte/protos/had_orc8r.proto]: can you instead make this mapping from uint32 (from eNB ID to offloadstate)?

**Review by ulaskozat — CHANGES_REQUESTED** [review]
see the comment regarding the eNB serial number vs. ID.

**Comment by mpgermano** [comment]
Addressed comment

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**